### DISCUSSION TITLE
How can I disable comment printing inside code chunks?

### DISCUSSION THREAD

**Discussion by danielvartan** [quarto-dev/quarto-cli#7521] [Q&A] [answered]
### Description

Please note that I'm referring to comments inside the code chunk, not in the output.

### Code chunk

````
```{r}
#| fig-height: 4
#| fig-width: 6

# library(datasets)
# library(ggplot2)

ggplot2::ggplot(faithful, ggplot2::aes(x = eruptions, y = waiting)) +
  ggplot2::geom_point() +
  ggplot2::xlim(0.5, 6) +
  ggplot2::ylim(40, 110) +
  ggplot2::geom_density_2d_filled(alpha = 0.5) +
  ggplot2::geom_density_2d(linewidth = 0.25, colour = "black") +
  ggplot2::theme(legend.position = "none")
```
````
### PDF output (could also be other formats, like HTML):

![image](https://github.com/quarto-dev/quarto-cli/assets/29462472/03830e02-d583-4ba2-bc30-320d7f73a88e)

### Desired output

![image](https://github.com/quarto-dev/quarto-cli/assets/29462472/86f67cf7-261c-4458-876d-f44858a57eb1)

**Comment by danielvartan** [comment]
The [solution](https://stackoverflow.com/questions/44727110/removing-comments-from-printing-in-knitr-output) provided by @yihui, which involves  `knitr::opts_chunk$set(tidy = TRUE, tidy.opts = list(comment = FALSE))` , affects more than just the comment lines in the code. It can lead to several issues with the output.

Ideally, the solution should only remove lines that start with `#`'

**Reply by yihui** [reply]
> It can lead to several issues with the output.

May I ask what the issues are?

If one issue is to indent by 2 spaces instead of 4, you can set `tidy.opts=list(comment=FALSE, indent=2)`, or globally `options(formatR.indent = 2)`.

While it's technically possible to make **formatR** remove comments without touching the rest of code, I'd like to know more about the issues you mentioned first. Also, @cscheid's answer below is convincing to me.

BTW, a solution without using **formatR** is to use `echo` to remove certain lines/expressions, e.g.,

````md
```{r, echo=-(1:2)}
# abc
# def
iris |>
  subset(Sepal.Width > 4)
```
````

Doc: https://yihui.org/knitr/options/#text-output

**Reply by danielvartan** [reply]
Hi @yihui,

Besides code style, one issue I've experienced with PDF rendering is text overflow. It seems that `formatR` does not respect the 80-column rule.

`echo=-(1:2)` is a good workaround. Thank you!

![image](https://github.com/quarto-dev/quarto-cli/assets/29462472/6c029a88-dc05-4756-a855-8d3559a3194b)

**Reply by yihui** [reply]
For that specific problem, you may try `tidy.opts=list(comment=FALSE, indent=2, width.cutoff=I(80))` but the width is not guaranteed to be respected.

**Reply by danielvartan** [reply]
Thank you @yihui ! I will give it a try.

**Comment by cscheid** [comment, marked as answer]
You're not going to like my answer, but: you don't. Comments are there to be read by people, and that's what the code output is there to support. Code that's not meant to be executed really ought to live inside a separate block with eval: false, or some related tool. I promise you it's a bad habit to use comments for that, and we don't want to incentivize that.

**Reply by cscheid** [reply]
If you truly must, then you can use (eg) pre-render scripts to process the code cells yourself however you choose to, and then write those to .qmd files.

**Reply by mcanouil** [reply]
FYI, commenting code is a bad practice as said and is even highlighted as such in `lintr` for instance.
So, enforcing this should be of your own making.
<img width="966" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/ff74fc6a-5524-4371-bc29-1c0f3e54620b">

**Reply by danielvartan** [reply]
I agree. I don't usually comment my code, it's only for special cases.

But I understand why there is no built-in solution for that.

Thank you for the prompt responses @cscheid and @mcanouil !
